src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,723 @@
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * Function to flag "height" tag
+     * 
+     * @param object
+     *            building object
+     * @param tags
+     *            building tags
+     * @return CheckFlag based on cases
+     */
+    private Optional<CheckFlag> heightTagFlagLogic(final AtlasObject object,
+            final Map<String, String> tags)
+    {
+        final String heightTag = tags.get(HeightTag.KEY);
+
+        // Case 4: building has an invalid "height" tag
+        if (this.hasInvalidHeightTag(heightTag))
+        {
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(INSTRUCTION_FOUR, object.getOsmIdentifier())));
+        }
+
+        // Don't run statistics if building contains building=apartments tag (apartments seem to be
+        // showing up a lot as outliers)
+        if (!(tags.containsKey(BuildingTag.KEY)
+                && tags.get(BuildingTag.KEY).equalsIgnoreCase(BuildingTag.APARTMENTS.toString())))
+        {
+            try
+            {
+
+                final Optional<Double> buildingHeightTagValue = this.parseHeightTag(heightTag);
+
+                if (buildingHeightTagValue.isPresent())
+                {
+                    final Optional<Tuple4<String, Double, Double, Double>> rectangleStatsWhichIntersectObjectOptional = this
+                            .getRectangleStatsWhichIntersectObject(this.storedAreasWithStatistics,
+                                    object, HeightTag.KEY);
+                    // Case 5: Building intersects stored area AND "height" tag is an outlier
+                    // compared
+                    // to surrounding buildings with "height" tag.
+                    if (rectangleStatsWhichIntersectObjectOptional.isPresent()
+                            && this.isOutlier(buildingHeightTagValue.get(),
+                                    rectangleStatsWhichIntersectObjectOptional.get().getSecond(),
+                                    rectangleStatsWhichIntersectObjectOptional.get().getThird(),
+                                    rectangleStatsWhichIntersectObjectOptional.get().getFourth()))
+                    {
+                        return Optional.of(this.createFlag(object, this.getLocalizedInstruction(
+                                INSTRUCTION_THREE, object.getOsmIdentifier())));
+                    }
+
+                    // Case 5: Building does not intersect stored area but "height" tag is an
+                    // outlier
+                    // compared to surrounding buildings with "height tag.
+                    if (rectangleStatsWhichIntersectObjectOptional.isEmpty())
+                    {
+                        return this.getStatisticsWithNewStoreEntry(object,
+                                buildingHeightTagValue.get(), HeightTag.KEY);
+                    }
+                }
+
+            }
+            catch (final Exception ignored)
```

Comment:
what can be thrown here? Can we use a more specific Exception subclass if possible and log the error?

Reply:
Actually can probably remove this try/catch section since the parsing double try/catch is within the parseHeightTag function.